doc/gen_support_matrix_html.rs

```diff
@@ -0,0 +1,426 @@
+        found: String,
+        board: String,
+    },
+    #[error("invalid functionality name `{found}` for chip `{chip}`")]
+    InvalidFunctionalityNameChip {
+        found: String,
+        chip: String,
+    },
+    #[error("invalid support key name `{found}` for functionality `{functionality}` for board `{board}`")]
+    InvalidSupportKeyNameBoard {
+        found: String,
+        functionality: String,
+        board: String,
+    },
+    #[error("invalid support key name `{found}` for functionality `{functionality}` for chip `{chip}`")]
+    InvalidSupportKeyNameChip {
+        found: String,
+        functionality: String,
+        chip: String,
+    },
+    #[error("missing support info on board `{board}` or chip `{chip}` regarding functionality `{functionality}`")]
+    MissingSupportInfo {
+        board: String,
+        chip: String,
+        functionality: String,
+    },
+    #[error("could not write the output HTML file `{path}`")]
+    WritingOutputFile {
+        path: PathBuf,
+        source: io::Error,
+    },
+    #[error("could not read existing output HTML file `{path}`")]
+    ReadingExistingFile {
+        path: PathBuf,
+        source: io::Error,
+    },
+    #[error("existing HTML file `{path}` is not up to date")]
+    ExistingHtmlNotUpToDate {
+        path: PathBuf,
+    },
+}
+
+mod schema {
+    use std::collections::HashMap;
+
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Debug, Serialize, Deserialize)]
+    #[serde(deny_unknown_fields)]
+    pub struct Matrix {
+        pub support_keys: HashMap<String, SupportKeyInfo>,
+        pub functionalities: Vec<FunctionalityInfo>,
+        pub chips: HashMap<String, ChipInfo>,
+        pub boards: HashMap<String, BoardInfo>,
+    }
+
+    #[derive(Debug, Serialize, Deserialize)]
+    #[serde(deny_unknown_fields)]
+    pub struct SupportKeyInfo {
+        pub icon: String,
+        pub description: String,
+    }
+
+    #[derive(Debug, Serialize, Deserialize)]
+    #[serde(deny_unknown_fields)]
+    pub struct FunctionalityInfo {
+        pub name: String,
+        pub title: String, // FIXME: rename this
+        pub description: String,
+    }
+
+    #[derive(Debug, Serialize, Deserialize)]
+    #[serde(deny_unknown_fields)]
+    pub struct ChipInfo {
+        pub name: String,
+        pub description: Option<String>,
+        pub support: HashMap<String, SupportInfo>,
+    }
+
+    #[derive(Debug, Serialize, Deserialize)]
+    #[serde(deny_unknown_fields)]
+    pub struct BoardInfo {
+        pub name: String,
+        pub description: Option<String>,
+        pub chip: String,
+        pub support: HashMap<String, SupportInfo>,
+    }
+
+    #[derive(Debug, Serialize, Deserialize)]
+    #[serde(deny_unknown_fields)]
+    #[serde(untagged)]
+    pub enum SupportInfo {
+        StatusOnly(String),
+        Details {
+            status: String,
+            comments: Option<String>,
+            github_pr: Option<u32>,
```

Comment:
In case you rework that for some reason or other, maybe just use a URI String here? The values are copy-pasted over from a browser anyway (I for one do not trust myself with copying over 3-digit numbers correctly), and that'd open up for having an issue, a PR, milestone item or even blog post in there, and it's one fewer hardcoded tie to GitHub.